crates/vault-registry/src/lib.rs

```diff
@@ -641,7 +647,16 @@ impl<T: Config> Pallet<T> {
             Self::is_allowed_to_withdraw_collateral(vault_id, amount)?,
             Error::<T>::InsufficientCollateral
         );
-
+        let vault_collateral = Self::get_vault_collateral(vault_id)?;
+        let backing_collateral = Self::get_backing_collateral(vault_id)?;
+        let current_nomination = backing_collateral.checked_sub(&vault_collateral).ok_or(Error::<T>::ArithmeticUnderflow)?;
+        let new_vault_collateral = vault_collateral
+            .checked_sub(&amount)
+            .ok_or(Error::<T>::ArithmeticUnderflow)?;
+        let max_nomination_after_withdrawal = Self::get_max_nominatable_collateral(new_vault_collateral)?;
+        if current_nomination.gt(&max_nomination_after_withdrawal) {
+            ext::staking::force_refund::<T>(&vault_id)?;
```

Comment:
I'm not sure we should do this here as well as on `opt_out_of_nomination`, especially since this wouldn't check secure collateralization.

Reply:
This is here to preserve the Max Nomination Ratio when the vault withdraws. It is different from the secure collateralization

Reply:
With respect to opting out, a vault needs to have the ability to deregister and thus return nominations. It was also a requirement in the design doc.

Reply:
Is it possible for the Vault to be undercollateralized by this call? This is also not defined in the spec.

Reply:
It may undercollateralize it, though not by a lot, due to the Max Nomination Ratio. We can warn the vault in the UI but they should have this option, to prevent having their collateral locked-in by nominators (which are untrusted).

Reply:
cc @nud3l

Reply:
Not sure I understand why it's useful here *as well* though, if we already have the opt-out function in the nomination pallet?

Reply:
Does make sense to only keep it there. And if the vault does become undercollateralized by opting out, at least they can't withdraw on top of that.

Reply:
However it would mean that the Max Nomination Ratio would no longer be an invariant. Are we ok with that?

Reply:
I think that is preferable, but if anything I would like to Error on withdrawal if the Max Nomination Ratio is no longer satisfied.